openedx/core/djangoapps/user_api/management/commands/migrate_user_profile_langs.py

```diff
@@ -0,0 +1,102 @@
+"""
+Migrates user preferences from one language code to another in batches. Dark lang preferences are not affected.
+"""
+from __future__ import print_function
+
+import logging
+from time import sleep
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.db.models import Q, Max
+
+from openedx.core.djangoapps.dark_lang.models import DarkLangConfig
+from openedx.core.djangoapps.user_api.models import UserPreference
+
+
+DEFAULT_CHUNK_SIZE = 10000
+DEFAULT_SLEEP_TIME_SECS = 10
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Implementation of the migrate command
+    """
+    help = 'Migrate all user language preferences (excluding dark languages) from one language code to another.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('old_lang_code',
+                            help='Original language code, ex. "zh-cn"')
+        parser.add_argument('new_lang_code',
+                            help='New language code, ex. "zh-hans"')
+        parser.add_argument('--start_id',
+                            type=int,
+                            default=1,
+                            help='ID to begin from, in case a run needs to be restarted from the middle.')
+        parser.add_argument('--chunk_size',
+                            type=int,
+                            default=DEFAULT_CHUNK_SIZE,
+                            help='Number of users whose preferences will be updated per batch.')
+        parser.add_argument('--sleep_time_secs',
+                            type=int,
+                            default=DEFAULT_SLEEP_TIME_SECS,
+                            help='Number of users whose preferences will be updated per batch.')
+
+    def handle(self, *args, **options):
+        """
+        Execute the command.
+        """
+        old_lang_code = options['old_lang_code']
+        new_lang_code = options['new_lang_code']
+        chunk_size = options['chunk_size']
+        sleep_time_secs = options['sleep_time_secs']
+        start = options['start_id']
+        end = start + chunk_size
+
+        # Make sure we're changing to a code that actually exists. Presumably it's safe to move away from a code that
+        # doesn't.
+        langs = [lang_code[0] for lang_code in settings.LANGUAGES]
+        langs += DarkLangConfig.current().released_languages_list
+
+        if new_lang_code not in langs:
+            raise CommandError('{} is not a configured language code in settings.LANGUAGES '
+                               'or the current DarkLangCongig.'.format(new_lang_code))
```

Comment:
"DarkLangCongig" -> "DarkLangConfig".  Also, won't the `format()` fail because it only operates on the second line's string?

Reply:
> Also, won't the format() fail because it only operates on the second line's string?

Seems that python's parser concatenates the string first, then applies the format to the full concatenated string.